roles/ncn-common/files/scripts/metal/metal-lib.sh

```diff
@@ -196,29 +202,36 @@ function set_static_fallback() {
     local ipaddr
     local lan
     local netmask
-    local netconf=/tmp/netconf
+    local vendor
 
     # BMCs either run dedicated on lan3 (last LAN channel as is the case with Intel's),
     # or lan1 (when there's only one channel).
-    if ipmi_output_3=$(ipmitool lan print 3 2>/dev/null); then
-        lan=3
-        echo "$ipmi_output_3" > $netconf
-    elif [ -z $lan ] && ipmi_output_1=$(ipmitool lan print 1 2>/dev/null); then
-        lan=1
-        echo "$ipmi_output_1" > $netconf
-    elif [ -z $lan ]; then
-        echo "Failed to determine which LAN channel to use!"
+    printf 'scanning vendor ... '
+    vendor=$(ipmitool fru | grep -i 'board mfg' | tail -n 1 | cut -d ':' -f2 | tr -d ' ')
+    echo "$vendor"
+    case $vendor in
+        *'Intel'*'Corporation'*)
+            lan=3
+            ;;
+        *)
+            lan=1
+            ;;
+    esac
+    ipaddr=$(host $(hostname)-mgmt | awk '{print $NF}')
+    netmask=$(cidr_to_netmask $(craysys metadata get --level node ipam | jq -r '.hmn.ip' | awk -F / '{print $NF}'))
+    defgw=$(craysys metadata get --level node ipam | jq -r '.hmn.gateway')
```

Comment:
If the cloud-init data is missing the hmn.gateway field, then defgw will end up being the string "null".   We should handle this case to avoid running the ipmitool command with the string "null".